logdetective/server.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import os
+import json
+
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+import requests
+
+from logdetective.constants import PROMPT_TEMPLATE
+from logdetective.extractors import DrainExtractor
+
+
+class BuildLog(BaseModel):
+    """Model of data submitted to API.
+    """
+    url: str
+
+LOG = logging.getLogger("logdetective")
+
+app = FastAPI()
+
+LLM_CPP_SERVER_ADDRESS = os.environ.get("LLAMA_CPP_SERVER", " http://localhost")
+LLM_CPP_SERVER_PORT = os.environ.get("LLAMA_CPP_SERVER_PORT", 8000)
+LLM_CPP_SERVER_TIMEOUT = os.environ.get("LLAMA_CPP_SERVER_TIMEOUT", 200)
+
+@app.post("/analyze", )
+async def analyze_log(build_log: BuildLog):
```

Comment:
took me so much time to figure out how the POST data are meant to look...

Reply:
That's a good point. I'll add some explanation in docstring.